File: Project Euler/12.cpp
#include <iostream>
using namespace std;

int divTriNum(int n) {
    int num = 0;
    int natNum = 1;
    for (;;) {
        num += natNum;
        natNum++;
        int nDivs = 2;
        for (int i = 2; i < num; i++) {
            if (not (num % i)) nDivs++;
            if (nDivs > n) {
                return num;
            }
        }
    }
}

int main(int argc, char const *argv[]){
    int num = 500;
    cout << divTriNum(num);
    return 0;
}

File: Project Euler/12.py
def div_tri_numbers(n):
    num = 0
    nat_num = 1
    while True:
        num += nat_num
        nat_num += 1
        n_divs = 2
        for i in range(2, num):
            if not num % i:
                n_divs += 1
            if n_divs == n:
                return num


def main():
    num = 500
    print(div_tri_numbers(num))


if __name__ == "__main__":
    main()
